plugin-annotation/src/main/java/com/mapbox/mapboxsdk/plugins/annotation/AnnotationManager.java

```diff
@@ -370,12 +372,18 @@ public void onDestroy() {
     private void initializeSourcesAndLayers(GeoJsonOptions geoJsonOptions) {
         geoJsonSource = coreElementProvider.getSource(geoJsonOptions);
         layer = coreElementProvider.getLayer();
-
         style.addSource(geoJsonSource);
-        if (belowLayerId == null) {
-            style.addLayer(layer);
-        } else {
+
+        if (belowLayerId != null && aboveLayerId != null) {
+            throw new RuntimeException("At most one of belowLayerId and aboveLayerId can be set, not both!");
+        }
```

Comment:
@fynngodau: Did you mean something like this?

Reply:
@JonasVautherin Yes, that's what I had in mind!

`IllegalArgumentException` could be a better fit? :thinking: